doc/developer-notes.md

```diff
@@ -793,25 +793,6 @@ bool ChainstateManager::ProcessNewBlock(...)
 }
 ```
 
-- When Clang Thread Safety Analysis is unable to determine if a mutex is locked, use `LockAssertion` class instances:
```

Comment:
In commit "Remove unused LockAssertion struct" (0bd1184adf6610c0bd14f4e9a25c0a200e65ae25)

Instead of removing this section, maybe it would have been better to update. The example could be updated with EXCLUSIVE_LOCKS_REQUIRED to be consistent with the code. And the text could be something like "When Clang Thread Safety Analysis is unable to determine if a mutex is locked, it is recommended to disable the analysis or to call the code through a function annotated with EXCLUSIVE_LOCKS_REQUIRED, using an indirect call so the requirement is not checked."

Reply:
Imo using `NO_THREAD_SAFETY_ANALYSIS` (or `thread_safety_cast` above) or `EXCLUSIVE_LOCKS_REQUIRED` on the lambda is equivalent, as the resulting binary, as well as any compile time errors/warnings should be the same.

As there is no observable difference (except in the written code itself), is there a reason to mention it in the dev notes?

Reply:
Imo, they're both not ideal. EXCLUSIVE_LOCKS_REQUIRED is preferable to NO_THREAD_SAFETY_ANALYSIS because it makes which mutex is assumed to be held explicit, and it doesn't disable checking for other mutexes. You might need to assume cs_main is held somewhere, and still want errors if you forget LOCK(cs_wallet) in the same function.

If it was useful before to have a section in developer notes advising how to deal with TSA errors, it's probably more useful now that the current practice is to add annotations that look like they might be enforced, but aren't enforced. But I don't know if the section was useful before it was removed. Just wanted to suggest here that it could be updated if it was useful.

Reply:
> You might need to assume cs_main is held somewhere, and still want errors if you forget LOCK(cs_wallet) in the same function.

Good point. Yeah, I agree the new "style" could be mentioned in the docs.